forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -237,3 +254,94 @@ impl WalletCommands {
         }
     }
 }
+
+fn formating_vars(balance_string: String, balance_exact: String) -> (String, String) {
+    if balance_exact.len() <= balance_string.len() {
+        let res = if balance_exact.ends_with(".0") {
+            //unfallible unwrap
+            balance_exact.strip_suffix(".0").unwrap().to_string()
+        } else {
+            balance_exact
+        };
+        (res, "".to_string())
+    } else {
+        //unfallible unwrap
+        let re = Regex::new(r"(\.0+|\.)$").unwrap();
+        let res = re.replace(balance_string.as_str(), "").to_string();
+        (res, "~".to_string())
+    }
+}
+
+fn format_balance_string(
+    mut balance_int: TokenAmount,
+    fixed_unit: &bool,
+    exact_balance: &bool,
+) -> String {
+    let mut unit = "FIL";
+    let (balance_string, symbol) = if *fixed_unit {
+        if *exact_balance {
+            formating_vars(format!("{balance_int}"), format!("{balance_int}"))
+        } else {
+            formating_vars(format!("{balance_int:.0}0"), format!("{balance_int}"))
+        }
+    } else {
+        let atto = balance_int.atto();
+        if *atto < BigInt::from(1000) {
+            unit = "atto FIL";
+            balance_int *= BigInt::from(1000000000000000000i64);
+        } else if *atto < BigInt::from(1000000) {
+            unit = "femto FIL";
+            balance_int *= BigInt::from(1000000000000000i64);
+        } else if *atto < BigInt::from(1000000000) {
+            unit = "pico FIL";
+            balance_int *= BigInt::from(1000000000000i64);
+        } else if *atto < BigInt::from(1000000000000i64) {
+            unit = "nano FIL";
+            balance_int *= BigInt::from(1000000000);
+        } else if *atto < BigInt::from(1000000000000000i64) {
+            unit = "micro FIL";
+            balance_int *= BigInt::from(1000000);
+        } else if *atto < BigInt::from(1000000000000000000i64) {
+            unit = "milli FIL";
+            balance_int *= BigInt::from(1000);
+        }
```

Comment:
There are a few problems with this code:

1. `balance_int` has type `TokenAmount` which is denominated in FIL but you're using it for other denominations. This is a conceptual type error.
2. You're using hard-coded values that are very error-prone. It would be easy to accidentally get `1000000000000000000i64` wrong. I'm sure this can be done in a way that's less error-prone and more readable.
3. The code is fairly repetitive. Each branch in the if-else chain does nearly the same thing.